Sources/Parsley/Parsley.swift

```diff
@@ -7,13 +7,22 @@ public enum MarkdownError: Error {
 
 public struct Parsley {
   /// This parses a String into HTML, without parsing Metadata or the document title.
-  public static func html(_ content: String, options: MarkdownOptions = [.safe]) throws -> String {
+  public static func html(
+    _ content: String,
+    options: MarkdownOptions = [.safe],
+    additionalSyntaxExtensions: [UnsafeMutablePointer<cmark_syntax_extension>] = []
```

Comment:
I don’t think we should add `UnsafeMutablePointer<cmark_syntax_extension>` to the public api, this needs to be more Swifty and less “unsafe”. Can you create an enum to wrap these, or something like that?

Reply:
And maybe the default value should be those default extensions that we register below, so that it's possible to actually override that. Now you can only add to it, but never prevent one of them from being registered. Not sure when you'd want that, but if we have the param anyway, might be nice. What do you think?

Reply:
> I don’t think we should add `UnsafeMutablePointer<cmark_syntax_extension>` to the public api, this needs to be more Swifty and less “unsafe”. Can you create an enum to wrap these, or something like that?

Wrapping it in a proper Swift API would be quite difficult because custom syntax extensions are pretty complex and it's mostly easiest just to copy one of the existing extension's implementations and modify it. If syntax extensions had to be made in Swift that wouldn't be possible. Also, some of the parser types don't seem to make it into Swift (such as `inline_parser`) which means that some extra C helpers would have to be made.

As an example use-case: I wanted to add syntax for custom modules like so: `@TableOfContents{depth=1}@` that gets transformed to `<module meta-name="TableOfContents" meta-parameters="{depth=1}">` which my html post processor can then render (TableOfContents is just an example). To implement that syntax I copied the strikethrough extension from cmark-gfm and modified it slightly. I tried porting the extension to Swift but it required the missing `inline_parser` API.

It would probably be possible to wrap the whole extension API in a nice and ergonomic Swift API (which I love the idea of), but I don't quite have the time at the moment (I'll let you know if I ever get around to trying it though).

I can do something like this though to hide the unsafe pointer from regular users if you want:

```swift
typealias SyntaxExtensionPointer = UnsafeMutablePointer<syntax_extension>

public enum SyntaxExtension {
  case strikethrough
  case autolink
  // ...
  case custom(SyntaxExtensionPointer)

  internal var pointer: SyntaxExtensionPointer {
    switch self {
      // ...
      case let .custom(pointer):
        return pointer
    }
  }
}
```

Another alternative (which I'm also fine with) is I just keep custom syntax extensions as a feature in my fork and don't merge it into yours, that way no one else has to see them, because I think probably no-one else using Parsley would want to make custom syntax extensions.

Reply:
> And maybe the default value should be those default extensions that we register below, so that it's possible to actually override that. Now you can only add to it, but never prevent one of them from being registered. Not sure when you'd want that, but if we have the param anyway, might be nice. What do you think?

Yep that's probably a good idea, I'll do that too

Reply:
That enum is exactly what I had in mind as well 👍